_posts/2021-04-07-upgrade-improvements.md

```diff
@@ -0,0 +1,71 @@
+In last releases we introduced several improvements to make the process easier.
+This post shows you how the upgrade process has changed, and what you should do to keep your cluster working without issues.
+
+<!--more-->
+
+### Upgrade improvements
+
+In the past, the upgrade process required multiple steps.
+During upgrade, users had to go through all released versions between the current version and the target version.
+For example if you wanted to upgrade from 0.18 to 0.20, the recommended approach was to install 0.19 first, wait for a rolling update, and set new values for Kafka `version`, `log.message.format.version` and `inter.broker.protocol.version`. 
+And then repeat the same steps for 0.20.
+This made the upgrades hard to execute fully automatically. For example, when installing Strimzi from the [OperatorHub](https://operatorhub.io/operator/strimzi-kafka-operator).
+The rolling update of Kafka cluster was also time-consuming, as upgrading through several versions can take a long time.
+
+#### Process changes
+
+Now, the process is a bit more friendly.
+The main improvement is you don't need to upgrade across all Strimzi versions, For instance, you can upgrade from version `X` directly to `X+2` without upgrade through any other release like `X+1`.
+The upgrade process is shown in the diagram below.
+
+Strimzi now does the following things regarding Kafka configuration:
+* Detect used Kafka `version`, `log.message.format.version` and `inter.broker.protocol.version` (from CR or compute it from Kafka `version`).
+* Configure Kafka `version`, `log.message.format.version` and `inter.broker.protocol.version` if it's not specified by user.
+
+Here's how Strimzi handles certain situations:
+* Supported Kafka `version` and `log.message.format.version` or `inter.broker.protocol.version` is set in CR - Strimzi will keep the values and missing one will se based on `version`.
+* Supported Kafka `version` is set, `log.message.format.version` and `inter.broker.protocol.version` are not set in CR - Strimzi will set `log.message.format.version` and `inter.broker.protocol.version` based on `version`
+* Supported Kafka `version` is not set in CR - Strimzi will set Kafka `version`, `log.message.format.version` and `inter.broker.protocol.version` based on the default Kafka supported by Strimzi.
+* Unsupported Kafka `version` is set in CR - Strimzi will set `Kafka` CR status to `NotReady`. 
+  In that case user has to update CR and manually set a proper config.
+
+You don't need to configure the Kafka `version`, `log.message.format.version` or `inter.broker.protocol.version` before the upgrade, but we **strongly** recommend it.
+
+#### Upgrade without versions
+
+Strimzi is able to deal with upgrade of Kafka clusters where versions are not set in the custom resource.
+In cases like this, Strimzi determines the versions and performs the upgrade as follows:
+* Get the version of the currently deployed Kafka and set it in the custom resource.
```

Comment:
Does it really set it in the CR?

Reply:
I removed that line with incorrect info.